src/08-recommendations/03-pipeline-recommend.ts

```diff
@@ -0,0 +1,325 @@
+const accessToken = process.env.RECOMMENDATIONS_USER_ACCESS_TOKEN
+
+if (!userID || !accessToken) {
+  throw new Error(
+    "Please provide a RECOMMENDATIONS_USER_ID and corresponding RECOMMENDATIONS_USER_ACCESS_TOKEN in the .env file to run this experiment"
+  )
+}
+
+// Settings
+
+/** fetch this many candidates from each of the recommendation fields */
+const NUM_CANDIDATES_PER_FIELD = 10
+
+/** applies to the `artworksForUser` field */
+const INCLUDE_BACKFILL = true
+
+/** for detailed logging of traffic to/from LLM and MP */
+const VERBOSE = false
+
+// Prompts
+
+const userDescription = `
+  John P. Has a collection of Print and Photography including works by Nathaniel Mary Quinn and Ewa Juszkiewicz, with a preference for Limited Edition works.
+`
+
+const systemPrompt = `
+  Your task is to recommend three artworks from a larger list of recommendation candidates. You should return the artworks that are most likely to match the user description. ONLY include artworks that are for sale. Prefer works that display a price.
+
+  If you receive no candidates, say that you have no candidates artworks to choose from.
+
+  Always make sure that the medium type matches the user's request.
+
+  Always include a justification as to why you are recommending each artwork.
+
+  Your output should include the artwork title, details as well as a one to two sentence description as to why this artwork matches the user's description.
+
+  Each of your artwork recommendations should have the following format including the —————— separator. Use plaintext only, no Markdown:
+
+  ——————————
+
+  ARTWORK: "<artwork title>", <artwork date>
+
+  - <artist name>, <artist nationality>, b. <artist birthday if present>
+  - <rarity> <medium>, <materials>
+  - slug: <slug>
+
+  WHY? <justification>
+`
+
+let candidateArtworks: { slug: string }[] = []
+
+// Main
+
+async function main() {
+  console.log(
+    chalk.bold(
+      `Fetching recommendations for user ${userID} and matching to the user description:\n`
+    ),
+    chalk.bold.blue(userDescription)
+  )
+
+  // initial message list
+
+  const messages: OpenAI.Chat.ChatCompletionMessageParam[] = [
+    { role: "system", content: systemPrompt },
+    { role: "user", content: initialUserMessage() },
+  ]
+
+  // first response
+
+  const firstResponse = await getCompletion(messages)
+
+  // second response, based on tool call
+
+  if (isTool(firstResponse)) {
+    messages.push(firstResponse.choices[0].message)
+
+    const toolResult = await getToolResult(firstResponse)
+    messages.push(toolResult)
+  }
+
+  const secondResponse: OpenAI.Chat.Completions.ChatCompletion | undefined =
+    await getCompletion(messages)
+
+  // final result
+
+  if (secondResponse !== undefined) {
+    const { content } = secondResponse.choices[0].message
+    console.log(chalk.bold("\nFinal response"))
+    console.log("\n", chalk.bold.blue(content))
+
+    printStatistics(content)
+  }
+}
+
+const initialUserMessage = () => `
```

Comment:
Total (non-blocking) nit, but do we want to be consistent with arrow function vs function expression use?

Reply:
Yep that didn't even need to be a function